src/backend/mpi/swdsm.cpp

```diff
@@ -312,19 +314,18 @@ void handler(int sig, siginfo_t *si, void *unused){
 
 	unsigned long tag;
 	argo_byte owner,state;
-	unsigned long distrAddr =  (unsigned long)((unsigned long)(si->si_addr) - (unsigned long)(startAddr));
-
-	unsigned long alignedDistrAddr = alignAddr(distrAddr);
-	unsigned long remCACHELINE = alignedDistrAddr % (CACHELINE*pagesize);
-	unsigned long lineAddr = alignedDistrAddr - remCACHELINE;
-	unsigned long classidx = get_classification_index(lineAddr);
-
-	unsigned long * localAlignedAddr = (unsigned long *)((char*)startAddr + lineAddr);
-	unsigned long startIndex = getCacheIndex(lineAddr);
-	alignedDistrAddr = lineAddr;
-
-	unsigned long homenode = getHomenode(lineAddr);
-	unsigned long offset = getOffset(lineAddr);
+	/* compute offset in distributed memory in bytes, always positive */
+	std::size_t access_offset = static_cast<char*>(si->si_addr) - static_cast<char*>(startAddr);
+
+	/* align access offset to cacheline */
+	std::size_t aligned_access_offset = align_backwards(access_offset, CACHELINE*pagesize);
+	unsigned long classidx = get_classification_index(aligned_access_offset);
+
+	/* compute start pointer of cacheline. char* has byte-wise arithmetics */
+	char* aligned_access_ptr = reinterpret_cast<char*>(startAddr) + aligned_access_offset;
+	unsigned long startIndex = getCacheIndex(aligned_access_offset);
+	unsigned long homenode = getHomenode(aligned_access_offset);
+	unsigned long offset = getOffset(aligned_access_offset);
 	unsigned long id = 1 << getID();
 	unsigned long invid = ~id;
```

Comment:
Eventually we should ensure that _all_ variables have better descriptive and more consistent names, in addition to using snake_case in accordance with the code standard. Not important for this specific PR, but the simple fact that `id` equals `1 << getID()` rather than simply the return value of `getID()` is not straightforward.

Reply:
Sure, eventually I would like to get rid of everything in this file and replace it with more modular, easier to read, better documented, and more consistent code. Should we create another ticket for this particular complaint, in the hope someone addresses it?

Reply:
I believe it is out of the scope of this PR, as changing either of the variable names above would lead to changes on multiple locations in the backend. Sounds like a good idea for another ticket to be dealt with in the future.

Reply:
This is now issue #36, please feel free to contribute your find there.